tests/agents/gherkin-specs/azure_functions_metadata.feature

```diff
@@ -0,0 +1,70 @@
+      | FUNCTIONS_EXTENSION_VERSION | version                                                                 |
+      | WEBSITE_OWNER_NAME          | d2cd53b3-acdc-4964-9563-3f5201556a81+faas_group-CentralUSwebspace-Linux |
+      | WEBSITE_SITE_NAME           | site_name                                                               |
+      | REGION_NAME                 | Central US                                                              |
+      | WEBSITE_RESOURCE_GROUP      | faas_group_from_env                                                     |
+    When cloud metadata is collected
+    Then cloud metadata is not null
+    And cloud metadata 'account.id' is 'd2cd53b3-acdc-4964-9563-3f5201556a81'
+    And cloud metadata 'provider' is 'azure'
+    And cloud metadata 'service.name' is 'functions'
+    And cloud metadata 'instance.name' is 'site_name'
+    And cloud metadata 'project.name' is 'faas_group_from_env'
+    And cloud metadata 'region' is 'Central US'
+
+  Scenario: WEBSITE_OWNER_NAME environment variable not expected format
+    Given the following environment variables are present
+      | name               | value                                                                   |
+      | WEBSITE_OWNER_NAME | d2cd53b3-acdc-4964-9563-3f5201556a81-faas_group-CentralUSwebspace-Linux |
+      | WEBSITE_SITE_NAME  | site_name                                                               |
+    When cloud metadata is collected
+    Then cloud metadata is null
+
+  Scenario: Missing FUNCTIONS_EXTENSION_VERSION environment variable
+    Given the following environment variables are present
+      | name               | value                                                                   |
+      | WEBSITE_OWNER_NAME | d2cd53b3-acdc-4964-9563-3f5201556a81+faas_group-CentralUSwebspace-Linux |
+      | WEBSITE_SITE_NAME  | site_name                                                               |
+    When cloud metadata is collected
+    Then cloud metadata is null
+
+  Scenario: Missing WEBSITE_OWNER_NAME environment variable
+    Given the following environment variables are present
+      | name                        | value     |
+      | FUNCTIONS_EXTENSION_VERSION | version   |
+      | WEBSITE_SITE_NAME           | site_name |
+    When cloud metadata is collected
+    Then cloud metadata is null
+
+  Scenario: Missing WEBSITE_SITE_NAME environment variable
+    Given the following environment variables are present
+      | name                        | value                                                                   |
+      | FUNCTIONS_EXTENSION_VERSION | version                                                                 |
+      | WEBSITE_OWNER_NAME          | d2cd53b3-acdc-4964-9563-3f5201556a81+faas_group-CentralUSwebspace-Linux |
+    When cloud metadata is collected
+    Then cloud metadata is null

```

Comment:
Is the intent here that Azure Functions instrumentation -- or at least *metadata* collection for Azure Functions -- is disabled if these envvars aren't defined? Currently the Node.js APM agent is gating solely on the presence of `FUNCTIONS_WORKER_RUNTIME`.

Reply:
Yes, this is the minimal information needed for Azure functions metadata collection.
Azure Functions *tracing* (it's not an implementation in the .NET case e.g.) is not impacted by that.

Reply:
I'm a little confused, but perhaps that is fine.  For the Node.js APM agent, both metadata collection and tracing are gated on the same check: the presence of the `FUNCTIONS_WORKER_RUNTIME` envvar.  Basically it is assuming that if `FUNCTIONS_WORKER_RUNTIME` is defined, then (mostly) the other envvars are defined.  I'm going to leave it at that for now -- unless someone thinks that is problematic.  (FWIW the Node.js APM agent doesn't currently use the gherkin spec files for testing.)

Reply:
In the .NET agent, these two aspects of Azure Functions are basically independent because:
* There is no actual instrumentation but a Middleware that customers *can* use to trace the function invocation. However, they could also just use the public API and achieve the same thing (though much more cumbersome). For that reason, the Middleware is not really gated by the presence of any environment variables. You only get inferior data if you would run it in a non-Azure-Functions environment.
* Metadata collection, on the other hand,  is a thing of its own that would report the correct "habitat information" if the agent was running in an Azure Function App w/o the middleware being used.

I hope that explains why I also tried to separate these concerns here in the spec.

Reply:
Is that tracing middleware on by default if the user has enabled the .NET APM agent?

Reply:
The user has to reference a dedicated package (`Elastic.Apm.Azure.Functions`) in their app and add a line of code to enable it. So, it's really a manual instrumentation approach.

Reply:
Thanks. I'm obviously pretty ignorant about .NET-land. :)  The coming docs (https://apm-agent-dotnet_1981.docs-preview.app.elstc.co/guide/en/apm/agent/dotnet/master/setup-azure-functions.html) help a lot.